Build failure with utfcpp >=3.0

Some deprecated functions have been deleted in utfcpp 3.0:
https://github.com/nemtrif/utfcpp/commit/00d73e4a02f98e8b993c222d6b741daf723fd23b

Ledger fails to build with utfcpp >=3.0:
```
.../ledger/src/textual.cc: In member function ‘std::streamsize ledger::{anonymous}::instance_t::read_line(char*&)’:
.../ledger/src/textual.cc:322:39: error: ‘is_bom’ is not a member of ‘utf8’
     if (context.linenum == 0 && utf8::is_bom(context.linebuf)) {
                                       ^~~~~~
.../ledger/src/textual.cc:322:39: note: suggested alternative: ‘bom’
     if (context.linenum == 0 && utf8::is_bom(context.linebuf)) {
                                       ^~~~~~
                                       bom
```

--- a/src/textual.cc
+++ b/src/textual.cc
@@ -319,7 +319,9 @@ std::streamsize instance_t::read_line(char *& line)
     context.curr_pos  = context.line_beg_pos;
     context.curr_pos += len;
 
-    if (context.linenum == 0 && utf8::is_bom(context.linebuf)) {
+    if (context.linenum == 0 &&
+        utf8::starts_with_bom(
+          context.linebuf, context.linebuf + sizeof(context.linebuf))) {
       line = &context.linebuf[3];
       len -= 3;
     } else {

